src/state_circuit/state.rs

```diff
@@ -0,0 +1,1609 @@
+};
+use pasta_curves::arithmetic::FieldExt;
+
+/*
+Example state table:
+
+| q_target | address | address_diff | global_counter | value | flag | padding |
+-------------------------------------------------------------------------------
+|    1     |    0    |              |       0        |  0    |   1  |    0    |  // init row (write value 0)
+|    2     |    0    |              |       12       |  12   |   1  |    0    |
+|    2     |    0    |              |       24       |  12   |   0  |    0    |
+|    2     |    1    |              |       0        |  0    |   1  |    0    |  // init row (write value 0)
+|    2     |    1    |              |       2        |  12   |   0  |    0    |
+|          |         |              |                |       |      |    1    |  // padding
+|          |         |              |                |       |      |    1    |  // padding
+|    1     |    0    |              |       3        |  4    |   1  |    0    |  // first stack op at the address has to be write
+|    3     |    0    |              |       17       |  32   |   1  |    0    |
+|    3     |    0    |              |       89       |  32   |   0  |    0    |
+|    3     |    1    |              |       48       |  32   |   1  |    0    |  // first stack op at the address has to be write
+|    3     |    1    |              |       49       |  32   |   0  |    0    |
+|          |         |              |                |       |      |    1    |  // padding
+*/
+
+// q_target:
+// 1 - first row of either target (Note: only the first row, not all init rows)
+// 2 - memory
+// 3 - stack
+// 4 - storage
+
+// address_diff is needed only internally to check whether the address changed
+// padding specifies whether the row is just a padding to fill all the rows that are intended
+// for a particular target
+
+/*
+Example bus mapping:
+// TODO: this is going to change
+
+| target | address | global_counter | value | flag |
+----------------------------------------------------
+|    2   |    0    |       12       |  12   |   1  |
+|    2   |    0    |       24       |  12   |   0  |
+|    2   |    1    |       2        |  12   |   0  |
+|    1   |    0    |       3        |  4    |   1  |
+|    3   |    0    |       17       |  32   |   1  |
+|    3   |    0    |       89       |  32   |   0  |
+|    3   |    1    |       48       |  32   |   1  |
+|    3   |    1    |       49       |  32   |   0  |
+*/
+
+/// In the state proof, memory operations are ordered first by address, and then by global_counter.
+/// Memory is initialised at 0 for each new address.
+/// Memory is a word-addressed byte array, i.e. a mapping from a 253-bit word -> u8.
+/// TODO: Confirm if we are using 253- or 256-bit memory addresses.
+#[derive(Copy, Clone, Debug)]
+struct Address<F: FieldExt>(F);
+
+/// Global counter
+#[derive(Copy, Clone, Debug)]
+struct GlobalCounter(usize);
+
+/// TODO: In the EVM we can only read memory values in 32 bytes, but can write either
```

Comment:
I think this point need to confirm, reading memory values is not fixed 32 except MLOAD op code,  others read values base on actual size parameter.
Refer to https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go